internal/handler/handler.go

```diff
@@ -0,0 +1,99 @@
+package handler
+
+import (
+	"fmt"
+	"github.com/Kuart/metric-collector/internal/metric"
+	"github.com/Kuart/metric-collector/internal/storage"
+	"github.com/Kuart/metric-collector/internal/template"
+	"github.com/go-chi/chi/v5"
+	"net/http"
+	"strconv"
+)
+
+const (
+	notIntError     = "value is not int type"
+	notFloatError   = "value is not float type"
+	metricTypeError = "metric type not implemented"
+	metricNotFound  = "metric \"%s\" is not found"
+)
+
+func SetRoutes(r *chi.Mux) {
+	r.Get("/value/{type}/{name}", MetricValueHandler)
+	r.Route("/update", func(r chi.Router) {
+		r.Post("/{type}/{name}/{value}", UpdateHandler)
```

Comment:
а можно жестко прописать:
``
```suggestion
		r.Post("/gauge/{name}/{value}", GaugeHandler)
		r.Post("/counter/{name}/{value}", CounterHandler)
```
тогда вообще можно избавиться от UpdateHandler 2 строчки VS 10 :)